src/openloco/s5/s5.h

```diff
@@ -0,0 +1,112 @@
+#pragma once
+
+#include "../objects/objectmgr.h"
+#include <cstdint>
+
+namespace openloco::s5
+{
+#pragma pack(push, 1)
+    struct Header
+    {
+        std::byte var_0;
+        std::byte var_1;
+        uint16_t var_2;
+        uint32_t var_4;
+        uint32_t var_8;
+        std::byte pad_C[20];
+    };
+#pragma pack(pop)
+    static_assert(sizeof(Header) == 0x20);
+
+#pragma pack(push, 1)
+    struct Options
+    {
+        uint8_t editor_step;        // 0x00
+        uint8_t difficulty;         // 0x01
+        uint16_t scenarioStartYear; // 0x02
+        uint8_t pad_4[2];
+        uint16_t scenarioFlags; // 0x06
+        uint8_t madeAnyChanges; // 0x08
+        uint8_t pad_9[1];
+        uint8_t landDistributionPatterns[32];           // 0x0A
+        char scenarioName[64];                          // 0x2A
+        char scenarioDetails[256];                      // 0x6A
+        objectmgr::header scenarioText;                 // 0x16A
+        uint16_t numberOfForests;                       // 0x17a
+        uint8_t minForestRadius;                        // 0x17C
+        uint8_t maxForestRadius;                        // 0x17D
+        uint8_t minForestDensity;                       // 0x17E
+        uint8_t maxForestDensity;                       // 0x17F
+        uint16_t numberRandomTrees;                     // 0x180
+        uint8_t minAltitudeForTrees;                    // 0x182
+        uint8_t maxAltitudeForTrees;                    // 0x183
+        uint8_t minLandHeight;                          // 0x184
+        uint8_t topographyStyle;                        // 0x185
+        uint8_t hillDensity;                            // 0x186
+        uint8_t numberOfTowns;                          // 0x187
+        uint8_t maxTownSize;                            // 0x188
+        uint8_t numberOfIndustries;                     // 0x189
+        uint8_t preview[128][128];                      // 0x18A
+        uint8_t copy_maxCompetingCompanies;             // 0x418A
```

Comment:
If these are copies, where are the originals?

Reply:
See `0x004401F7`

Reply:
Looking at the xrefs to the other locations makes me think those are the copies instead. They seem to be used for some very local purposes there, while the ones in this struct are used in many more places.

I'd prefer to remove the `copy_` prefix here.

Reply:
![image](https://user-images.githubusercontent.com/737603/59311675-b32e5d80-8caa-11e9-9ff9-88cacdc26afa.png)

These copies are all used to set up the scenario index. `sub_444C4E` loads them into the actual addresses, but that only seems to be to format the objective description.

Reply:
Indeed, I had found as much in IDA myself. That's perfectly fine, though, right? I'd expect those functions to work directly with the data as it exists in scenario files.

To me, there is a difference between the scenario as it exists on file and the game-instance variables. I don't expect the game to keep using these `s5` structs in-game — you've demonstrated they get overwritten from other variables at some point.

That does not change my mind about the prefix, though. It really seems to be that the values from these structs are copied to do some local work, not the other way around. Neither of which warrant a `copy_` prefix, to me, not in the least because the instance-variables can exist in a different namespace.